tests/Unit/MartinGeorgiev/Doctrine/DBAL/Types/ValueObject/LtreeTest.php

```diff
@@ -0,0 +1,526 @@
+        yield 'list with null' => [['a', null, 'c']];
+        yield 'list with dotted string' => [['a', 'b.c', 'ds']];
+    }
+
+    #[DataProvider('provideInvalidStringRepresentation')]
+    #[Test]
+    public function throws_exception_for_invalid_string_representation(string $value): void
+    {
+        $this->expectException(InvalidLtreeException::class);
+        Ltree::fromString($value);
+    }
+
+    /**
+     * @return iterable<string, string[]>
+     */
+    public static function provideInvalidStringRepresentation(): iterable
+    {
+        yield 'string starting with dot' => ['.b'];
+        yield 'string ending with dot' => ['a.'];
+    }
+
+    /**
+     * @param list<non-empty-string> $expected
+     */
+    #[DataProvider('provideValidRepresentation')]
+    #[Test]
+    public function can_create_from_string(string $value, array $expected): void
+    {
+        $ltree = Ltree::fromString($value);
+        self::assertSame($expected, $ltree->getPathFromRoot());
+        self::assertSame($value, (string) $ltree);
+    }
+
+    /**
+     * @param list<non-empty-string> $value
+     */
+    #[DataProvider('provideValidRepresentation')]
+    #[Test]
+    public function can_convert_to_string(string $expected, array $value): void
+    {
+        $ltree = new Ltree($value);
+        self::assertSame($expected, (string) $ltree);
+    }
+
+    /**
+     * @param list<non-empty-string> $expected
+     */
+    #[DataProvider('provideValidRepresentation')]
+    #[Test]
+    public function can_serialize_to_json(string $value, array $expected): void
+    {
+        $ltreeFromString = Ltree::fromString($value);
+        self::assertSame($expected, $ltreeFromString->jsonSerialize());
+
+        $ltree = new Ltree($expected);
+        self::assertSame($expected, $ltree->jsonSerialize());
+    }
+
+    /**
+     * @return iterable<string, array{0: string, 1: list<non-empty-string>}>
+     */
+    public static function provideValidRepresentation(): iterable
+    {
+        yield 'empty string' => ['', []];
+        yield 'single node' => ['a', ['a']];
+        yield 'multiple nodes' => ['a.b.c', ['a', 'b', 'c']];
+        yield 'with numbers' => ['1.2.3', ['1', '2', '3']];
+        yield 'with special characters' => ['a.b.c-d_e', ['a', 'b', 'c-d_e']];
+    }
+
+    #[Test]
+    public function can_encode_to_json_array(): void
+    {
+        $ltree = new Ltree(['a', 'b', 'c']);
+        $json = \json_encode($ltree, \JSON_THROW_ON_ERROR);
+        self::assertSame('["a","b","c"]', $json);
+    }
+
+    #[DataProvider('provideParentRelationship')]
+    #[Test]
+    public function can_get_parent(Ltree $child, Ltree $parent): void
+    {
+        $ltree = $child->getParent();
+        self::assertSame((string) $parent, (string) $ltree);
+    }
+
+    #[DataProvider('provideParentRelationship')]
+    #[Test]
+    public function respect_immutability_when_getting_parent(Ltree $child, Ltree $parent): void
```

Comment:
I don't understand the purpose of this test or its intended assertions intentions. The parent is not used at all, but it is passed as an argument. Can you please elaborate on them?